nd2reader/parser.py

```diff
@@ -247,6 +247,28 @@ def _channel_offset(self):
         """
         return {channel: n for n, channel in enumerate(self.metadata["channels"])}
 
+    def _check_unwanted_bytes(self, image_group_data, image_data_start, height, width):
+        number_of_true_channels = int(len(image_group_data[4:]) / (height * width))
+        n_unwanted_bytes = (len(image_group_data[image_data_start:]))%(height*width)
+        if not n_unwanted_bytes:
+            return False
+        assert 0 == n_unwanted_bytes % height, "Unexpected unwanted bytes distribution."
```

Comment:
Maybe make the assertion message a bit more descriptive here, something like: "An unexpected number of extra bytes was encountered based on the expected frame size, therefore the file could not be parsed."